src/shell/types.rs

```diff
@@ -632,6 +658,81 @@ impl KillSignalDropGuard {
   }
 }
 
+#[derive(Debug, Default)]
+struct ProcessSignalerInner {
+  /// The PID of the current foreground process, if any.
+  current_pid: Cell<Option<u32>>,
+  /// Sender for process spawn notifications.
+  /// Lazily initialized on first subscribe.
+  sender: RefCell<Option<broadcast::Sender<u32>>>,
+}
+
+/// Provides access to the currently running foreground child process.
+///
+/// This is useful for signal forwarding scenarios where you need to check
+/// if the child process is in the same process group as the parent.
+///
+/// # Example
+///
+/// ```ignore
+/// let signaler = ProcessSignaler::new();
+/// let mut receiver = signaler.subscribe();
+///
+/// // In a signal handler, check if we should forward the signal
+/// if let Some(child_pid) = signaler.current_pid() {
+///     // Check if child is in same process group
+///     let child_pgid = unsafe { libc::getpgid(child_pid as i32) };
+///     let our_pgid = unsafe { libc::getpgid(0) };
+///
+///     if child_pgid != our_pgid {
+///         // Child in different process group, forward signal
+///         kill_signal.send(SignalKind::SIGINT);
+///     }
```

Comment:
I don't think this would work because there could be multiple child processes running in different groups?